token-metadata-creator/src/Cardano/Metadata/Types.hs

```diff
@@ -158,10 +161,23 @@ newtype Property = Property { unProperty :: Text }
 hashProperty :: Property -> Hash Blake2b_256 Property
 hashProperty = hashWith (CBOR.toStrictByteString . CBOR.encodeString . unProperty)
 
+newtype Decimals = Decimals { unDecimals :: Int }
+  deriving (Eq, Show)
+
+instance WellKnownProperty Decimals where
+  wellKnownPropertyName _ =
+    Property "decimals"
+  wellKnownToBytes =
+    CBOR.encodeInt . unDecimals
+  wellKnownToJSON =
+    toJSON . unDecimals
+  parseWellKnown =
+    Aeson.withScientific "decimals" validateMetadataDecimals
```

Comment:
It would be better to avoid any use of the Scientific type.
Something like this ought to work...
```suggestion
  parseWellKnown = parseJSON >=> validateMetadataDecimals
```